signer/src/eth.rs

```diff
@@ -0,0 +1,346 @@
+    let pub_key = secp256k1::PublicKey::from_slice(&pub_key.0).expect("valid public key");
+
+    Secp256k1::verification_only()
+        .verify_ecdsa(&wrapped, &signature, &pub_key)
+        .is_ok()
+}
+
+/// An error handed back if creating the keypair fails.
+#[derive(Debug, PartialEq, Display, From)]
+pub enum Error {
+    /// Invalid private key.
+    #[display(fmt = "Invalid private key")]
+    #[from(ignore)]
+    InvalidPrivateKey,
+    /// Invalid hex.
+    #[display(fmt = "Cannot parse hex string: {_0}")]
+    Hex(hex::FromHexError),
+    /// Inner,
+    #[display(fmt = "{_0}")]
+    Inner(ecdsa::Error),
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for Error {}
+
+/// Dev accounts, helpful for testing but not to be used in production,
+/// since the secret keys are known.
+pub mod dev {
+    use super::*;
+    use core::str::FromStr;
+
+    once_static_cloned! {
+        /// Equivalent to `{DEV_PHRASE}//Alice`.
+        pub fn alice() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Alice").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Bob`.
+        pub fn bob() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Bob").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Charlie`.
+        pub fn charlie() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Charlie").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Dave`.
+        pub fn dave() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Dave").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Eve`.
+        pub fn eve() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Eve").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Ferdie`.
+        pub fn ferdie() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Ferdie").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//One`.
+        pub fn one() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//One").unwrap()).unwrap()
+        }
+        /// Equivalent to `{DEV_PHRASE}//Two`.
+        pub fn two() -> Keypair {
+            Keypair::from_uri(&SecretUri::from_str("//Two").unwrap()).unwrap()
+        }
+    }
+}
+
+#[cfg(feature = "subxt")]
+mod subxt_compat {
+    use super::*;
+
+    impl<T: subxt::Config> subxt::tx::Signer<T> for Keypair
+    where
+        T::AccountId: From<AccountId20>,
+        T::Address: From<AccountId20>,
+        T::Signature: From<Signature>,
+    {
+        fn account_id(&self) -> T::AccountId {
+            self.account_id().into()
+        }
+
+        fn address(&self) -> T::Address {
+            self.account_id().into()
+        }
+
+        fn sign(&self, signer_payload: &[u8]) -> T::Signature {
+            self.sign(signer_payload).into()
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
I reckon it's worth also copying the moonbeam tests here to do a sanity chekc that our impl lines up with theirs:

https://github.com/moonbeam-foundation/moonbeam/blob/e70ee0d427dfee8987d5a5671a66416ee6ec38aa/primitives/account/src/lib.rs#L217

(For the other impls we can sanity check against substrate, but here we want to sanity check that we align with eth chains)